src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -138,60 +149,107 @@ const TableHeader: React.FC<TableHeaderProps> = ({
     };
   }, [leaveSeat]);
 
+  const updateTableTitle = useCallback(
+    async (newTableTitle) => {
+      if (!user) return;
+
+      const venueTables = tables.map((table) => {
+        if (table.reference === tableOfUser?.reference) {
+          return { ...table, title: newTableTitle };
+        }
+        return table;
+      });
+
+      // Ideally we want to do this only when config.tables doesn't exist. Otherwise we want to update only a single table.
+      await updateVenue_v2(
+        { name: venueName, tables: venueTables },
+        user
+      ).catch(() => {
+        // @debt Replace generic error when proper error handling is added
+        setError(GENERIC_ERROR);
+
+        const msg = "[updateVenue_v2] updating tables in venue.config.tables";
+        const context = {
+          location: "molecules::TableHeader::updateTableTitle",
+        };
+
+        console.warn(msg, context);
+        Bugsnag.notify(msg, (event) => {
+          event.severity = "warning";
+          event.addMetadata("context", context);
+        });
+      });
+    },
+    [tableOfUser?.reference, tables, user, venueName]
+  );
+
   return (
-    // @debt move all style into TableHeader.scss
-    <div className="row no-margin at-table table-header">
-      <div className="header">
-        <div className="back-button-container">
-          <button
-            type="button"
-            title={"Leave " + seatedAtTable}
-            className="back-button"
-            id="leave-seat"
-            onClick={leaveSeat}
-          >
-            <FontAwesomeIcon
-              className="back-button-icon"
-              icon={faChevronLeft}
-              size="xs"
-            />
-            Leave table
-          </button>
-        </div>
-        <div className="table-title-container">
-          <div className="private-table-title">
-            {tableTitle}
-
-            {tableCapacity && (
-              <span className="private-table-seats-left">
-                {numberOfSeatsLeft} seats left
-              </span>
-            )}
-          </div>
-          {tableOfUser && tableOfUser.subtitle && (
-            <div className="private-table-subtitle">{tableOfUser.subtitle}</div>
-          )}
-        </div>
-        <div className="lock-button-container">
+    <div className="row table-header">
+      <div className="table-header__leave-table">
+        <button
+          type="button"
+          title={"Leave " + seatedAtTable}
+          className="table-header__leave-button"
+          id="leave-seat"
```

Comment:
Do we need this `id` here? `id`s aren't very Reacty.

Reply:
Nope, I will remove this.

Reply:
Removed in: https://github.com/sparkletown/sparkle/pull/1377/commits/de659957079dd74136f26d29e9b952be12add510